sunpy/cube/cube.py

```diff
@@ -0,0 +1,477 @@
+# -*- coding: utf-8 -*-
+# Author: Mateo Inchaurrandieta <mateo.inchaurrandieta@gmail.com>
+# pylint: disable=E1101, E0611
+"""
+Main class for representing cubes - 3D sets of continuous data by time and/or
+wavelength
+"""
+# NOTE: This module uses version 1.02 of "Time coordinates in FITS" by
+# Rots et al, available at http://hea-www.cfa.harvard.edu/~arots/TimeWCS/
+# This draft standard may change.
+
+# standard libraries
+import datetime
+
+# external libraries
+import numpy as np
+import matplotlib.pyplot as plt
+import astropy.nddata
+from astropy import units as u
+from astropy.units import sday  # sidereal day
+
+# Sunpy modules
+from sunpy.map import GenericMap, MapMeta
+from sunpy.visualization.imageanimator import ImageAnimator
+from sunpy.lightcurve import LightCurve
+from sunpy.spectra.spectrum import Spectrum
+from sunpy.spectra.spectrogram import Spectrogram
+from sunpy.cube import cube_utils as cu
+from sunpy.wcs import wcs_util as wu
+
+__all__ = ['Cube']
+
+
+class Cube(astropy.nddata.NDDataArray):
```

Comment:
The way to deal with direct data slicing is with the slicing mixin http://docs.astropy.org/en/stable/nddata/index.html#mixins-for-additional-functionality.

I would be wary of using that though, because of some of the discussions we had about slicing last year. (Many of which I have thought about a lot more since!)